src/go/k8s/api/redpanda/v1alpha2/user_types.go

```diff
@@ -0,0 +1,196 @@
+	// +kubebuilder:validation:item:Enum=Read;Write;Delete;Alter;Describe;IdempotentWrite;ClusterAction;Create;AlterConfigs;DescribeConfigs
+	// +kubebuilder:validation:item:MaxLength=15
+	// +kubebuilder:validation:item:UniqueItems=true
+	Operations []string `json:"operations"`
+}
+
+// ACLResourceSpec indicates the resource for which given ACL rule applies.
+// +kubebuilder:validation:XValidation:message="prefixed pattern type only supported for ['group', 'topic', 'transactionalId']",rule="self.type in ['group', 'topic', 'transactionalId'] ? true : !has(self.patternType) || self.patternType != 'prefixed'"
+// +kubebuilder:validation:XValidation:message="match pattern type only supported for ['group', 'topic', 'transactionalId']",rule="self.type in ['group', 'topic', 'transactionalId'] ? true : !has(self.patternType) || self.patternType != 'match'"
+// +kubebuilder:validation:XValidation:message="any pattern type only supported for ['group', 'topic', 'transactionalId']",rule="self.type in ['group', 'topic', 'transactionalId'] ? true : !has(self.patternType) || self.patternType != 'any'"
+// +kubebuilder:validation:XValidation:message="name must not be specified for patternType 'any'",rule=` has(self.patternType) && self.patternType == 'any' ? self.name == "" : true`
+// +kubebuilder:validation:XValidation:message="name must not be specified for type ['cluster']",rule=`self.type == "cluster" ? (self.name == "") : true`
+// +kubebuilder:validation:XValidation:message="acl rules on non-cluster resources must specify a name when not using patternType 'any'",rule=`self.type == "cluster" ? true : (self.name != "" || self.patternType == 'any')`
+type ACLResourceSpec struct {
+	// +kubebuilder:validation:Enum=topic;group;cluster;transactionalId
+	Type string `json:"type"`
+	// Name of resource for which given ACL rule applies.
+	// Can be combined with patternType field to use prefix pattern.
+	Name string `json:"name"`
+	// Describes the pattern used in the resource field. The supported types are literal
+	// and prefixed. With literal pattern type, the resource field will be used as a definition
+	// of a full topic name. With prefix pattern type, the resource name will be used only as
+	// a prefix. Default value is literal.
+	//
+	// +kubebuilder:validation:Enum=any;literal;match;prefixed
+	// +default:value=literal
+	PatternType *string `json:"patternType,omitempty"`
+}
+
+// UserStatus defines the observed state of a Redpanda user
+type UserStatus struct {
+	// Specifies the last observed generation.
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+	// Conditions holds the conditions for the Redpanda user.
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	// ClusterRef is a reference to the cluster where the user was created.
+	// This is used so that if a ClusterRef of a User is changed, we can
+	// properly clean up the User created in the previous cluster.
+	ClusterRef *ClusterRef `json:"clusterRef,omitempty"`
```

Comment:
I went ahead an made it immutable, adding in mutability later would be a non-breaking change anyway, so it feels like it'll be cleaner to handle things like that initially -- WDYT about doing the same for the `StaticConfiguration` (the thing holding `KafkaAPISpec`/`AdminAPISpec` fields? There's a bit of an implicit assumption that those never change at the moment, because, as soon as they do, we'd be unable to clean up any old values from a previously referenced cluster.

Reply:
SGTM! And even if one would want to change those, they can just delete the CR and recreate it as a workaround until we implement mutability. (Assuming someone asks for it).